app/main.py

```diff
@@ -1,3 +1,41 @@
-def shop_trip():
-    # write your code here
-    pass
+from _datetime import datetime
+import json
+
+from app.trips_cost import calculate_trips_cost, Shop
+
+
+def shop_trip() -> None:
+    with open("config.json", "r") as file:
+        config = json.load(file)
+        fuel_price = config["FUEL_PRICE"]
+        customers = config["customers"]
+        shops_config = config["shops"]
+        shops = [
+            Shop(shop["name"], shop["location"], shop["products"])
+            for shop in shops_config
+        ]
+        for customer in customers:
+            cheapest_cost, cheapest_shop, rest_of_money = calculate_trips_cost(
+                customer, shops, fuel_price
+            )
+            if rest_of_money >= 0:
+                print(f"{customer['name']} rides to {cheapest_shop[0]}")
+                date = datetime(2021, 4, 1, 12, 33, 41)
+                print(f"\nDate: {date.strftime('%m/%d/%Y %H:%M:%S')}\n"
```

Comment:
shouldnt the shop print the purchase check with current time according to the task? please check

Reply:
Unfortunately - not, to my surprise, the tests checked that the date would be a certain, from the readme